airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -143,6 +143,7 @@ def __init__(self, *args, **kwargs) -> None:
         self.schema = kwargs.pop("schema", None)
         self.authenticator = kwargs.pop("authenticator", None)
         self.session_parameters = kwargs.pop("session_parameters", None)
+        self.application = kwargs.pop("application")
```

Comment:
```suggestion
        self.application = kwargs.pop("application", None)
```

Reply:
Pushed a commit to fix this and other static checks: https://github.com/apache/airflow/pull/16420/commits/26b344f53b2b248a7b201bd4c775d8dc1b6974ee